### ISSUE TITLE
Database Help! My Parent isn't recognizing it's child

### ISSUE THREAD

**Issue by toryburgett** [ga-wdi-exercises/project3#119]
Hi GA

I have an issue. I have 2 models: EventPosts and Tldrs. EventPosts have many Tldrs (Tldrs should only belong to one EventPost). I am trying to create a "create" method on the front end which will create a new Tldr and attribute it to its correct EventPost.I am able to create new Tldrs, but I am not able to connect them to their parent EventPost. For context, here is my code:

controllers/EventPosts.js

```
//// Putting a new tldr in a eventPost

router.post("/:id/tldrs", function(req, res){
  // console.log(req.body);
  // var ep = EventPost.findByIdAndUpdate(req.params.id);
    var newTldr = new Tldr(req.body);
    newTldr.save().then(
      EventPost.findByIdAndUpdate(req.params.id, {$push: {tldrs: newTldr} })).then(function(tldr){
        res.json(tldr);
      });
});

///////////////
```

public/js/models/tldr.js

```
Tldr.create = function(info, eventPostID){
  var self = this;
  var url = "/eventposts/"+eventPostID+"/tldrs";
  var request = $.ajax({
    url: url,
    method: "post",
    data: JSON.stringify(info),
    contentType: 'application/json'
  });
  return request;
};
```

public/js/views/eventPostView.js

```
    createTldrButton.on("click", function(){
      $(".formNewTldr"+self.eventPost.id).show();
      var cancelCreateTldr = self.$el.find(".cancelCreateTldr"+self.eventPost.id);
      var createnewTldrButton = self.$el.find(".createnewTldrButton"+self.eventPost.id);
      cancelCreateTldr.on("click", function(){
        event.preventDefault();
        $(".formNewTldr"+self.eventPost.id).hide();
      });
      createnewTldrButton.on("click", function(){
        event.preventDefault();
        console.log("Create New Button Clicked! EventPost ID: "+self.eventPost.id);
        // creaet a new tldr, attatching it to the eventPost
        var data = {  summary: $(".summaryInputTldr"+self.eventPost.id).val(),
                        relevance: $(".relevanceInputTldr"+self.eventPost.id).val()
                      };
        var eventPostID = self.eventPost.id;
        Tldr.create(data, eventPostID).then(function(response){
          console.log(data);
          var tldr = new Tldr(response);
          console.log("creating a new Tldr");
          var view = new TldrView(tldr);
          console.log(view);
          view.render();
        });
          $(".formNewTldr"+self.eventPost.id).hide();
        createnewTldrButton.off();
      });
    });
```

**Comment by andrewsunglaekim** [comment]
are the `tldr`'s being saved as subdocuments? seems like they are.. Can you try:

```
var newTldr = new Tldr(req.body);
EventPost.findByIdAndUpdate(req.params.id, {$push: {tldrs: newTldr} })).then(function(tldr){
  res.json(tldr);
});
```

in your post method

**Comment by RobertAKARobin** [comment]
This issue's title is the story of my life. :(

**Comment by toryburgett** [comment]
THank you both, Jesse helped, the problem was in the controllers/eventPosts.js

```
router.post("/:id/tldrs", function(req, res){
  var tldr = new Tldr(req.body);
  tldr.save(function(d){
    EventPost.findByIdAndUpdate(req.params.id, {
      $push: {
        "tldrs": tldr
      }
    }).then(function(err,doc){
      res.json(tldr);
    });
  });
});
```

**Comment by toryburgett** [comment]
and also, Robin:  :(
